src/compiler/checker.ts

```diff
@@ -7254,14 +7253,37 @@ module ts {
                 }
             }
 
+            function isImportedNameFromExternalModule(n: Node): boolean {

```

Comment:
This is already an inner function so it should be able to refer to the node from the outer function. Why you have to pass it in as function argument?